driver process (ros_main) stops receiving but doesn't actually terminate when it gets signalled SIGINT

The driver node (process `ros_main`) only half-heartedly responds to SIGINT. 

**Steps to reproduce the behavior**:
Launch the driver, e.g. craft a `params.yaml` then start the binary directly on command line
```
cat <<EOT > /tmp/params.yaml
killtest:
  ros__parameters:
    transport: "udp"
    scanner_ip: "10.0.10.9"
    max_num_points_scan: 0
    frame_id: "scanner_link"
    device: "R2000"
    start_angle: -1800000
    packet_type: "C"
    scan_topic: "/scan"
    watchdogtimeout: 60000
    watchdog: true
EOT

$WS/install/pf_driver/lib/pf_driver/ros_main --ros-args -r __node:=killtest --params-file /tmp/params.yaml
```

Then try to kill it with Ctrl-C. It will receive SIGINT and change its state, but never terminates:
```
 ^Creceived SIGINT
[INFO] [1733938913.220970697] [killtest]: Device state changed to Initialized
[ERROR] [1733938913.221122614] [killtest]: Network failure
[ERROR] [1733938915.206894965] [transport]: Time out: No packets received in 2 seconds
^Creceived SIGINT
^Creceived SIGINT
^Creceived SIGINT
^Creceived SIGINT
^Creceived SIGINT
```
Although termination is not so important during normal use, it hinders execution of tests that start, stop, re-start the driver...

**Environment**:
Driver 1.3 and 2.0 (e.g. c1a4e12) in ROS Humble, Ubuntu 22.04 LTS

--- a/src/pf_driver/src/pf/pipeline.cpp
+++ b/src/pf_driver/src/pf/pipeline.cpp
@@ -89,9 +89,10 @@ void Pipeline::run_writer()
   running_ = false;
 
   // notify main thread about network failure
+  if (net_mtx_->try_lock())
   {
-    std::lock_guard<std::mutex> lock(*net_mtx_);
     net_fail_ = true;
+    net_mtx_->unlock();
   }
   net_cv_->notify_one();
 }

--- a/src/pf_driver/src/ros/ros_main.cpp
+++ b/src/pf_driver/src/ros/ros_main.cpp
@@ -177,32 +177,33 @@ int main(int argc, char* argv[])
   });
 
   if (rclcpp::ok())
-  do
   {
-    net_fail = false;
-    if (!pf_interface.init(info, config, params, topic, frame_id, num_layers))
+    do
     {
-      RCLCPP_ERROR(node->get_logger(), "Unable to initialize device");
-      if (retrying)
+      net_fail = false;
+      if (!pf_interface.init(info, config, params, topic, frame_id, num_layers))
       {
-        std::this_thread::sleep_for(std::chrono::milliseconds(2000));
-        continue;
+        RCLCPP_ERROR(node->get_logger(), "Unable to initialize device");
+        if (retrying)
+        {
+          std::this_thread::sleep_for(std::chrono::milliseconds(2000));
+          continue;
+        }
+        return -1;
       }
-      return -1;
-    }
-    if (!pf_interface.start_transmission(net_mtx_, net_cv_, net_fail))
-    {
-      RCLCPP_ERROR(node->get_logger(), "Unable to start scan");
-      return -1;
-    }
-    retrying = true;
-    // wait for condition variable
-    std::unique_lock<std::mutex> net_lock(*net_mtx_);
-    net_cv_->wait(net_lock, [] { return net_fail; });
-    RCLCPP_ERROR(node->get_logger(), "Network failure");
-    pf_interface.terminate();
+      if (!pf_interface.start_transmission(net_mtx_, net_cv_, net_fail))
+      {
+        RCLCPP_ERROR(node->get_logger(), "Unable to start scan");
+        return -1;
+      }
+      retrying = true;
+      // wait for condition variable
+      std::unique_lock<std::mutex> net_lock(*net_mtx_);
+      net_cv_->wait(net_lock, [] { return net_fail; });
+      RCLCPP_ERROR(node->get_logger(), "Network failure");
+      pf_interface.terminate();
+    } while (rclcpp::ok() && retrying);
   }
-  while (rclcpp::ok() && retrying);
 
   std::cout << "I am here" << std::endl;
   pf_interface.stop_transmission();

--- a/src/pf_driver/src/ros/ros_main.cpp
+++ b/src/pf_driver/src/ros/ros_main.cpp
@@ -7,6 +7,8 @@
 
 #include "pf_driver/pf/pf_interface.h"
 
+static volatile bool retrying = false;
+
 int main(int argc, char* argv[])
 {
   rclcpp::init(argc, argv);
@@ -153,7 +155,6 @@ int main(int argc, char* argv[])
   static std::shared_ptr<std::mutex> net_mtx_ = std::make_shared<std::mutex>();
   static std::shared_ptr<std::condition_variable> net_cv_ = std::make_shared<std::condition_variable>();
   static bool net_fail = false;
-  bool retrying = false;
 
   rclcpp::executors::MultiThreadedExecutor executor;
   executor.add_node(node);
@@ -167,6 +168,7 @@ int main(int argc, char* argv[])
       // notify main thread about network failure
       {
         std::lock_guard<std::mutex> lock(*net_mtx_);
+        retrying = false;
         net_fail = true;
       }
       net_cv_->notify_one();
@@ -174,7 +176,8 @@ int main(int argc, char* argv[])
     });
   });
 
-  while (rclcpp::ok())
+  if (rclcpp::ok())
+  do
   {
     net_fail = false;
     if (!pf_interface.init(info, config, params, topic, frame_id, num_layers))
@@ -199,6 +202,8 @@ int main(int argc, char* argv[])
     RCLCPP_ERROR(node->get_logger(), "Network failure");
     pf_interface.terminate();
   }
+  while (rclcpp::ok() && retrying);
+
   std::cout << "I am here" << std::endl;
   pf_interface.stop_transmission();
   rclcpp::shutdown();

